src/Core/Utilities/CoreHelpers.cs

```diff
@@ -810,5 +810,15 @@ public static T LoadClassFromJsonData<T>(string jsonData) where T : new()
 
             return System.Text.Json.JsonSerializer.Deserialize<T>(jsonData, options);
         }
+
+        public static ICollection<T> AddIfNotExists<T>(this ICollection<T> list, T item)
+        {
+            if (list.Contains(item))
+            {
+                return list;
+            }
+            list.Add(item);
+            return list;
+        }
```

Comment:
Isn't this what a HashSet is?

Reply:
Yes, that is exactly what a `HashSet<T>` does (to some degree, although it's way faster), but this was for dealing with general interface types like `ICollection<T>` of properties from classes we do not have control over (to simply code and reduce repetitive code). I can take this out of the extensions and just re-work the code in the `DynamicAuthenticationSchemeProvider` accordingly, not a big deal.